### ISSUE TITLE
REFERENCE field

### ISSUE THREAD

**Issue by kablag** [gschofl/biofiles#5]
Files without REFERENCE field raise
`Error in { : task 1 failed - "'to' cannot be NA, NaN or infinite"`
doe it necessary field?

**Comment by gschofl** [comment]
That shouldn't happen. Do you have an example of such a file?

**Comment by kablag** [comment]
[18S-r.zip](https://github.com/gschofl/biofiles/files/1495134/18S-r.zip)

```
gbRecord("18S-r.gb")
Error in { : task 1 failed - "'to' must be a finite number"
```
After adding random REFERENCE field
```
gbRecord("18S-r__2.gb")
An object of class ‘gbRecord’, with 1 features
LOCUS       18S-r                     18 bp    DNA     linear   UNA 20-APR-2017
DEFINITION  A new oligonucleotide entered manually.
ACCESSION   urn.local...3i-7h23ry9
VERSION     urn.local...3i-7h23ry9GI:NA
KEYWORDS    .
SOURCE      
  ORGANISM  
            .
REFERENCE   1  (bases 1 to 18)
  AUTHORS   Tristan, C, Shahani, N, Sedlak, TW, Sawa and A.
  TITLE     The diverse functions of GAPDH: views from different subcellular compartments
  JOURNAL   Cell. Signal. 23 (2), 317-323 (2011)
  PUBMED    20727968
  REMARK    Review article
ORIGIN      GSWCTATCCCCAKCACGA
CONTIG   
```
Maybe error isn't inside REFERENCE but function doesn't return any usefull information on error.

**Comment by tolot27** [comment]
Try to remove
```
SOURCE      
  ORGANISM  
            .
```
I believe the line with the single dot causes the error. In general, this GenBank record looks ugly to me, because ACCESSION and VERSION does not follow a common convention.

**Comment by kablag** [comment]
It does not helped for me.
Bad thing is that my file from example is the standard export from Geneious and i can't use such files directly...

**Comment by tolot27** [comment]
>  i can't use such files directly...

Why not? Just open the file with a text editor and remove the lines. Alternatively, set the toxonomic information, e. g. taxid 32630 (synthetic construct) in Geneious.

**Comment by tolot27** [comment]
But the real problem is more likely the wrong unconventional ACCESSION/VERSION. Typically, it is like `NC_002505.1`. Hence try to correct the accession and version of your entry to match this naming scheme.

**Comment by kablag** [comment]
[18S-r.zip](https://github.com/gschofl/biofiles/files/1495311/18S-r.zip)
Unfortunately I got the same error after adding taxid or removing lines.

**Comment by kablag** [comment]
> But the real problem is more likely the wrong unconventional ACCESSION/VERSION. Typically, it is like NC_002505.1. Hence try to correct the accession and version of your entry to match this naming scheme.

it does not help

**Comment by tolot27** [comment]
Try this:

```
LOCUS       18S-r                     18 bp    DNA     linear   UNA 22-NOV-2017
DEFINITION  A new oligonucleotide entered manually.
ACCESSION   7h23ry9
VERSION     7h23ry9.1
KEYWORDS    .
FEATURES             Location/Qualifiers
     primer_bind     1..18
ORIGIN      
        1 gswctatccc cakcacga
//
```

**Comment by kablag** [comment]
Does not help.
Is it work for you?

**Comment by gschofl** [comment]
So far I've treated the REFERENCE field in the GenBank file header as a mandatory element. That is why the parser failed on this example. There is no clear guidance from NCBI that REFERENCE has to be a mandatory field so I relaxed this assumption.

SOURCE with ORGANISM with the complete lineage are *still* mandatory, but leaving them empty as in 
```
SOURCE      
  ORGANISM  
            .
```
is fine, and works now as expected in the development version on GH
